src/decoder/lattice-incremental-decoder.cc

```diff
@@ -0,0 +1,1617 @@
+         iter != reprocess.end(); ++iter)
+      reprocess_vec.push_back(*iter);
+    reprocess.clear();
+    for (typename std::vector<Token *>::iterator iter = reprocess_vec.begin();
+         iter != reprocess_vec.end(); ++iter) {
+      Token *tok = *iter;
+      int32 pos = token2pos[tok];
+      // Repeat the processing we did above (for comments, see above).
+      for (ForwardLinkT *link = tok->links; link != NULL; link = link->next) {
+        if (link->ilabel == 0) {
+          IterType following_iter = token2pos.find(link->next_tok);
+          if (following_iter != token2pos.end()) {
+            int32 next_pos = following_iter->second;
+            if (next_pos < pos) {
+              following_iter->second = cur_pos++;
+              reprocess.insert(link->next_tok);
+            }
+          }
+        }
+      }
+    }
+  }
+  KALDI_ASSERT(loop_count < max_loop &&
+               "Epsilon loops exist in your decoding "
+               "graph (this is not allowed!)");
+
+  topsorted_list->clear();
+  topsorted_list->resize(cur_pos, NULL); // create a list with NULLs in between.
+  for (IterType iter = token2pos.begin(); iter != token2pos.end(); ++iter)
+    (*topsorted_list)[iter->second] = iter->first;
+}
+
+template <typename FST, typename Token>
+bool LatticeIncrementalDecoderTpl<FST, Token>::GetLattice(CompactLattice *olat) {
+  return GetLattice(true, NumFramesDecoded(), olat);
+}
+
+template <typename FST, typename Token>
+bool LatticeIncrementalDecoderTpl<FST, Token>::GetLattice(bool use_final_probs,
+                                                          int32 last_frame_of_chunk,
+                                                          CompactLattice *olat) {
+  using namespace fst;
+  bool not_first_chunk = last_get_lattice_frame_ != 0;
+  bool ret = true;
+
+  // last_get_lattice_frame_ is used to record the first frame of the chunk
+  // last time we obtain from calling this function. If it reaches
+  // last_frame_of_chunk
+  // we cannot generate any more chunk
+  if (last_get_lattice_frame_ < last_frame_of_chunk) {
+    Lattice raw_fst;
+    // step 1: Get lattice chunk with initial and final states
+    // In this function, we do not create the initial state in
+    // the first chunk, and we do not create the final state in the last chunk
+    if (!GetIncrementalRawLattice(&raw_fst, use_final_probs, last_get_lattice_frame_,
+                                  last_frame_of_chunk, not_first_chunk,
+                                  !decoding_finalized_))
+      KALDI_ERR << "Unexpected problem when getting lattice";
+    // step 2-3
+    ret = determinizer_.ProcessChunk(raw_fst, last_get_lattice_frame_,
+                                     last_frame_of_chunk);
+    last_get_lattice_frame_ = last_frame_of_chunk;
+  } else if (last_get_lattice_frame_ > last_frame_of_chunk)
+    KALDI_WARN << "Call GetLattice up to frame: " << last_frame_of_chunk
```

Comment:
there needs to be a bracket for the else if part because of there is one for the if part before (consistency according to Google style)